client/dashboard/me/billing-purchases/purchase-settings/upcoming-renewals-dialog.tsx

```diff
@@ -0,0 +1,141 @@
+		} );
+	}
+	// translators: "expiry" is relative to the present time and it is already localized, eg. "in a year", "in a month", "a week ago"
+	return sprintf( __( 'expires %(expiry)s' ), {
+		expiry: getRelativeTimeString( new Date( purchase.expiry_date ) ),
+	} );
+}
+
+export function UpcomingRenewalsDialog( {
+	siteDomain,
+	purchases,
+	onClose,
+	onConfirm,
+	submitButtonText,
+	showManagePurchaseLinks = true,
+}: Props ) {
+	const [ selectedPurchases, setSelectedPurchases ] = useState< number[] >( [] );
+
+	const purchasesSortByRecentExpiryDate = useMemo(
+		() =>
+			[ ...purchases ].sort( ( a, b ) => {
+				const compareDateA = isRenewing( a ) ? a.renew_date : a.expiry_date;
+				const compareDateB = isRenewing( b ) ? b.renew_date : b.expiry_date;
+				return compareDateA?.localeCompare?.( compareDateB );
+			} ),
+		[ purchases ]
+	);
+
+	useEffect( () => {
+		setSelectedPurchases( purchases.map( ( purchase ) => purchase.ID ) );
+	}, [ purchases ] );
+
+	return (
+		<ConfirmDialog
+			size="large"
+			confirmButtonText={ submitButtonText ?? __( 'Renew now' ) }
+			onConfirm={ () =>
+				onConfirm( purchases.filter( ( purchase ) => selectedPurchases.includes( purchase.ID ) ) )
+			}
+			onCancel={ onClose }
+		>
+			<VStack>
+				<Heading>{ __( 'Upcoming renewals' ) }</Heading>
+				<Text variant="muted">
+					{
+						// translators: siteName is the URL of the site
+						sprintf( __( 'Site: %(siteName)s' ), { siteName: siteDomain } )
+					}
+				</Text>
+			</VStack>
+			<Divider margin={ 3 } />
+			{ purchasesSortByRecentExpiryDate.map( ( purchase ) => {
+				const purchaseTypeText = getSubtitleForDisplay( purchase );
+				const onChange = () => {
+					if ( selectedPurchases.includes( purchase.ID ) ) {
+						setSelectedPurchases( selectedPurchases.filter( ( id ) => id !== purchase.ID ) );
+					} else {
+						setSelectedPurchases( selectedPurchases.concat( [ purchase.ID ] ) );
+					}
+				};
+				return (
+					<VStack key={ purchase.ID }>
+						<HStack>
+							<HStack alignment="left">
+								<CheckboxControl
+									name={ `${ purchase.product_slug }-${ purchase.ID }` }
+									checked={ selectedPurchases.includes( purchase.ID ) }
+									onChange={ onChange }
+								/>
+								<VStack>
+									<Text>{ purchase.is_domain ? purchase.meta ?? '' : purchase.product_name }</Text>
+									<Text variant="muted">
+										{ purchaseTypeText ? `${ purchaseTypeText }: ` : '' }
+										<span>{ purchaseTypeText && <ExpiresText purchase={ purchase } /> }</span>
+									</Text>
+								</VStack>
+							</HStack>
+							<VStack>
+								<Text>
+									{ formatCurrency(
+										purchase.sale_amount ?? purchase.amount,
+										purchase.currency_code,
+										{
+											stripZeros: true,
+										}
+									) }
+								</Text>
+								{ showManagePurchaseLinks && (
+									<div className="upcoming-renewals-dialog__renewal-settings-link">
```

Comment:
Would it be better to avoid it shrink? Also, it seems the class name is unused 🤔

Reply:
Ah, thanks. I missed this when copying from the original component.